eventlet/support/greendns.py

```diff
@@ -469,16 +473,19 @@ def _getaddrinfo_lookup(host, family, flags):
     addrs = []
     if family == socket.AF_UNSPEC:
         err = None
-        for qfamily in [socket.AF_INET6, socket.AF_INET]:
-            try:
-                answer = resolve(host, qfamily, False)
-            except socket.gaierror as e:
-                if e.errno not in (socket.EAI_AGAIN, EAI_NONAME_ERROR.errno, EAI_NODATA_ERROR.errno):
-                    raise
-                err = e
-            else:
-                if answer.rrset:
-                    addrs.extend(rr.address for rr in answer.rrset)
+        for use_dns in [False, True]:
+            for qfamily in [socket.AF_INET6, socket.AF_INET]:
+                try:
+                    answer = resolve(host, qfamily, False, use_dns=use_dns)
+                except socket.gaierror as e:
+                    if e.errno not in (socket.EAI_AGAIN, EAI_NONAME_ERROR.errno, EAI_NODATA_ERROR.errno):
+                        raise
+                    err = e
+                else:
+                    if answer.rrset:
+                        addrs.extend(rr.address for rr in answer.rrset)
+            if addrs:
+                break
         if err is not None and not addrs:
             raise err
     elif family == socket.AF_INET6 and flags & socket.AI_V4MAPPED:
```

Comment:
When `resolve()` family is specified, it works as before, going to network. Is that how `getaddrinfo` works also?

Reply:
There's another `else` block below too.

Reply:
When resolve() family is specified it works as before, ie. returning the entry on hosts file and not using network if it's there.
My change basically applies to getaddrinfo() where if a IPv4 or IPv6 entry was fetched from /etc/hosts it won't attempt to get it from the network. Before this patch, if for example only an IPv4 entry was found, it'll query the network for the IPv6 resolution. Makes sense?

Reply:
Yes, sorry, I meant `_getaddrinfo_lookup()` with family not AF_UNSPEC.

Reply:
Yes if family not AF_UNSPEC then my patch is not changing the behavior

Reply:
It should, to match system `getaddrinfo`, right?

Reply:
@temoto before my patch, individual calls to `resolve()` (ie. family not being AF_UNSPEC) where already checking hosts file first. Problem was that calls with AF_UNSPEC will call twice to resolve (one with AF_INET6 and one with AF_INET) and if only one (4 or 6) entry is present in hosts file, the other call will contact the network.